src/main/java/org/opensearch/neuralsearch/processor/NormalizationProcessor.java

```diff
@@ -131,4 +143,29 @@ private <Result extends SearchPhaseResult> Optional<FetchSearchResult> getFetchS
         Optional<Result> optionalFirstSearchPhaseResult = searchPhaseResults.getAtomicArray().asList().stream().findFirst();
         return optionalFirstSearchPhaseResult.map(SearchPhaseResult::fetchResult);
     }
+
+    private boolean shouldSkipProcessorDueToIncompatibleQueryAndFetchResults(
+        final List<QuerySearchResult> querySearchResults,
+        final Optional<FetchSearchResult> fetchSearchResultOptional
+    ) {
+        if (fetchSearchResultOptional.isEmpty()) {
+            return false;
+        }
+        final List<Integer> docIds = unprocessedDocIds(querySearchResults);
+        SearchHits searchHits = fetchSearchResultOptional.get().hits();
+        SearchHit[] searchHitArray = searchHits.getHits();
+        // validate the both collections are of the same size
+        if (Objects.isNull(searchHitArray) || searchHitArray.length != docIds.size()) {
+            return true;
+        }
```

Comment:
Lets add logs for this. Make it info logs so that we know what happened.

Reply:
ack